jetty-client/src/test/java/org/eclipse/jetty/client/Socks4ProxyTest.java

```diff
@@ -318,7 +323,9 @@ public void testIdleTimeoutWhenSocksProxyDoesNotRespond() throws Exception
             // Accept the connection, but do not reply and don't close.
 
             ExecutionException x = assertThrows(ExecutionException.class, () -> listener.get(2 * idleTimeout, TimeUnit.MILLISECONDS));
-            assertThat(x.getCause(), instanceOf(TimeoutException.class));
+            Class<?> expectedException = "ci".equals(System.getProperty("env")) ? ConnectException.class : TimeoutException.class;
```

Comment:
I definitely reproduce it locally 100% of the time
```
[WARN] [stderr] Caused by: 
[WARN] [stderr] java.net.ConnectException: Connection refused
[WARN] [stderr] 	at java.base/sun.nio.ch.Net.pollConnect(Native Method)
[WARN] [stderr] 	at java.base/sun.nio.ch.Net.pollConnectNow(Net.java:672)
[WARN] [stderr] 	at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:973)
[WARN] [stderr] 	at org.eclipse.jetty.io@10.0.13-SNAPSHOT/org.eclipse.jetty.io.SelectorManager.doFinishConnect(SelectorManager.java:341)
```

Reply:
Don't do this. Maybe open an issue to be investigated, but if you make this change it will be forgotten.
Likely a Jenkins/Docker/K8s issue because I cannot reproduce it locally.

Reply:
done